src/wampy.js

```diff
@@ -530,25 +530,20 @@ class Wampy {
     _validateURI (uri, isPatternBased, isWampAllowed) {
         const isStrictValidation = this._options.uriValidation === 'strict';
         const isLooseValidation = this._options.uriValidation === 'loose';
-        let reBase, rePattern;
 
         if (!isStrictValidation && !isLooseValidation) {
             return false;
         }
 
-        if (uri.startsWith('wamp.') && !isWampAllowed) {
+        if (uri?.startsWith('wamp.') && !isWampAllowed) {
             return false;
         }
 
-        if (isStrictValidation) {
-            reBase = /^([0-9a-zA-Z_]+\.)*([0-9a-zA-Z_]+)$/;
-            rePattern = /^([0-9a-zA-Z_]+\.{1,2})*([0-9a-zA-Z_]+)$/;
-        } else if (isLooseValidation) {
-            reBase = /^([^\s.#]+\.)*([^\s.#]+)$/;
-            rePattern = /^([^\s.#]+\.{1,2})*([^\s.#]+)$/;
-        }
+        const char = isStrictValidation ? '[0-9a-zA-Z_]' : '[^\\s.#]';
+        const dots = isPatternBased ? '\\.{1,2}' : '\\.';
+        const uriPattern = new RegExp(`^(${char}+${dots})*${char}+$`);
```

Comment:
Hm... I was looking at this a few times, and think that the original variant is a little easier to read/follow. Because here there are a few inline checks first, then Regexp is constructed via template string. So to figure out what flow will be — you have to look at regexp, destruct the template there, then holding that in mind try to compile in the head what values will be for each template parameter...

So at least, it is hard to construct and see the whole regexp. I don't talk about debugging.

Reply:
Hmm I see your point, my initial idea was to make the differences between each regex line easier to spot since they had a lot similarities. So by doing this it's easier to see that the char matcher and the dots matcher are the ones that are changing. But I agree the flow has been hurt. 

One alternative is to add comments with examples of URIs that match each type of validation, but if you feel better keeping the current version I'm fine with that 😃